nemo_skills/pipeline/generate.py

```diff
@@ -393,36 +552,52 @@ def generate(
                 chunk_id=None,
             )
         for chunk_id in chunk_ids:
-            # Configure client (same as before)
-            server_config, server_address, extra_arguments = pipeline_utils.configure_client(
-                model=model,
-                server_type=server_type,
-                server_address=original_server_address,
-                server_gpus=server_gpus,
-                server_nodes=server_nodes,
-                server_args=server_args,
-                server_entrypoint=server_entrypoint,
-                server_container=server_container,
-                extra_arguments=extra_arguments_original,
-                get_random_port=get_random_port,
-            )
-
-            # Build generation command (same as before)
-            cmd = pipeline_utils.get_generation_cmd(
-                input_file=input_file,
-                input_dir=input_dir,
-                random_seed=seed,
-                output_dir=output_dir,
-                extra_arguments=extra_arguments,
-                chunk_id=chunk_id,
-                num_chunks=num_chunks,
-                preprocess_cmd=preprocess_cmd,
-                postprocess_cmd=postprocess_cmd,
-                wandb_parameters=wandb_parameters if seed_idx == 0 else None,
-                script=generation_module,
-                with_sandbox=with_sandbox,
-            )
-            cmd = pipeline_utils.wrap_python_path(cmd=cmd)
+            server_configs = []
+            server_addresses_resolved = []
+
+            # For n=1, we need extra_arguments from configure_client
+            # For n>1, we build server addresses ourselves
+            extra_args_from_first_server = extra_arguments_original
+
+            for idx in range(num_models):
```

Comment:
consider using model_idx instead of idx to make this a bit clearer as this is already inside two loops with their own indices